objects/tag.c

```diff
@@ -139,7 +139,6 @@ tag_view(lua_State *L, int udx, bool view)
     {
         tag->selected = view;
         banning_need_update();
-        ewmh_update_net_current_desktop();
```

Comment:
Sorry, but I don't understand. What you are saying is:
- For simplicity: Screen 1 has tag 1-1, screen 2 has tag 2-1 and tag 2-2
- Chromium is on 1-1, tags 1-1 and 2-2 are selected
- Thus, `_NET_CURRENT_DESKTOP` is 1-1 currently (it contains the focused client)
- You deselect 2-2 and select 2-1 (switch to another tag on the second screen)
- `_NET_CURRENT_DESKTOP` does not change, because the focused client is still on 1-1 (well, awesome will set the value that this property already has again, but that does not matter)

Since `ewmh_update_net_current_desktop` does not look at the mouse pointer, mouse moves do not matter to it. What the function does look at is focused client and selected tag, thus it needs to connect to "focus" and "unfocus" on client (oh, unfocus is missing!), "tagged" and "untagged" also on client and "property::selected" on tags.